package.json

```diff
@@ -19,47 +20,52 @@
 	},
 	"devDependencies": {
 		"@babel/plugin-proposal-private-property-in-object": "^7.21.11",
+		"@babel/preset-typescript": "^7.24.6",
 		"@storybook/addon-essentials": "^8.1.1",
 		"@storybook/addon-interactions": "^8.1.1",
 		"@storybook/addon-links": "^8.1.1",
 		"@storybook/blocks": "^8.0.5",
-		"@storybook/preset-create-react-app": "^8.1.1",
 		"@storybook/react": "^8.1.1",
+		"@storybook/react-vite": "^8.1.3",
 		"@storybook/react-webpack5": "^8.1.1",
 		"@storybook/test": "^8.0.5",
 		"@testing-library/jest-dom": "^6.4.5",
 		"@testing-library/react": "^15.0.7",
 		"@testing-library/user-event": "^14.5.2",
-		"@types/jest": "^29.5.12",
 		"@types/node": "^18.19.33",
 		"@types/react": "^18.3.2",
 		"@types/react-dom": "^18.3.0",
+		"@typescript-eslint/eslint-plugin": "^7.10.0",
+		"@typescript-eslint/parser": "^7.10.0",
+		"@vitejs/plugin-react": "^4.3.0",
 		"babel-plugin-named-exports-order": "^0.0.2",
+		"eslint": "^v8.57.0",
 		"eslint-config-airbnb": "^19.0.4",
-		"eslint-config-airbnb-typescript": "^17.1.0",
+		"eslint-config-airbnb-typescript": "^18.0.0",
 		"eslint-config-prettier": "^9.1.0",
+		"eslint-plugin-react-hooks": "^4.6.2",
+		"eslint-plugin-react-refresh": "^0.4.7",
+		"jsdom": "^24.0.0",
 		"prettier": "^3.2.5",
-		"react-scripts": "5.0.1",
 		"storybook": "^8.1.1",
-		"typescript": "^4.9.5"
+		"typescript": "^5.2.10",
+		"vitest": "^1.6.0"
 	},
 	"scripts": {
-		"start": "react-scripts start",
+		"start": "vite",
 		"build": "npm run build:app && npm run build:storybook",
-		"build:app": "react-scripts build",
+		"build:app": "tsc && vite build",
 		"build:storybook": "storybook build -o build/storybook",
 		"storybook": "storybook dev -p 6006",
-		"test": "react-scripts test",
-		"lint:prettier": "prettier . --check",
-		"lint:eslint": "eslint ./src --ext .ts,.tsx,.js --max-warnings=0",
 		"lint": "npm run lint:eslint && npm run lint:prettier",
+		"lint:prettier": "prettier . --check",
+		"lint:eslint": "eslint . --ext ts,tsx,js --report-unused-disable-directives --max-warnings 0",
```

Comment:
This is now linting more than `/src` -- I know why we're doing this, but it's a change that could have accidental inclusions as a result so wanted to just point it out / make sure that running lint after build doesn't result in lint issues.

Reply:
Oh -- I did want to point out the addition of `--report-unused-disable-directives`

What is the reasoning for that?

Reply:
> Oh -- I did want to point out the addition of `--report-unused-disable-directives`
> 
> What is the reasoning for that?

According to the eslint docs, it's for reporting any disables that wouldn't have thrown errors anyway! https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives

Reply:
@hminsky2002 has changed the `lint:eslint` command to reference `./src` instead of `.`, and there should be no new errors from the unused disable directives, so it's okay to add.